.github/workflows/test2.yml

```diff
@@ -0,0 +1,42 @@
+name: example-workflow
+on:
+  push:
+    branches:
+      - testing-action
+jobs:
+  run-anchor-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+          cache: "npm"
+
+      - name: Install dependencies
+        run: |
+          npm i -g yarn
+          yarn install
+
+      - name: Set up Rust Toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+      - run: |
+          rustc --version
+          cargo --version
+
+      - name: Set up Anchor CLI
+        uses: metaDAOproject/setup-anchor@v2.1
+        with:
+          anchor-version: "0.30.1"
+          solana-cli-version: "1.18.25"
+          node-version: "20"
```

Comment:
Great decision to use a pre-defined action, instead of manually installing 👌 I can see that with this new action, overall time taken by the action has reduced dramatically.

We can take this a step further. According to [the docs](https://www.anchor-lang.com/docs/verifiable-builds#images), anchor already provides [official images](https://hub.docker.com/layers/backpackapp/build/v0.30.1/images/sha256-c160fe32cba7c463981110b2aac2924de4833c06a0af6473a830ead880c4ef3b?context=explore) that come with rust, solana CLI and anchor already installed. Can you see if this can be used as the [container image for our github action workflow](https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container)? It will eliminate the need to install these dependencies during action build time

Reply:
On second thought, I think [setup-node action](https://github.com/actions/setup-node) would be the fastest way to do it. It seems like this action takes ~3 seconds on avg (probably uses cache). Wdyt?